src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -651,6 +651,8 @@ private function writeProperty($zval, $property, $value)
             // fatal error.
 
             $object->$property = $value;
+        } elseif (self::ACCESS_TYPE_NOT_FOUND === $access[self::ACCESS_TYPE]) {
+            throw new NoSuchPropertyException(sprintf('Could not determine access type for property "%s".', $property));
         } elseif (self::ACCESS_TYPE_MAGIC === $access[self::ACCESS_TYPE]) {
```

Comment:
I'm sorry but this is not the same... all condition have to be checked before comming to that line, which means all previous conditions before have an impact on the conditions below. If you break the chain, you break the code...